api/tests/Behat/bootstrap/v2/UserManagement/UserManagementTrait.php

```diff
@@ -0,0 +1,677 @@
+    /**
+     * @When I can see the user as non active in search results
+     */
+    public function iCanSeeTheUserAsNonActiveInSearchResults()
+    {
+        $this->iAmOnAdminUsersSearchPage();
+        $xpath = '//tbody/tr';
+        $firstRow = $this->getSession()->getPage()->find('xpath', $xpath)->getHtml();
+        $this->assertStringContainsString($this->expectedUsers[0]['firstName'], $firstRow, 'Add user check - first name');
+        $this->assertStringContainsString($this->expectedUsers[0]['lastName'], $firstRow, 'Add user check - last name');
+        $this->assertStringContainsString($this->expectedUsers[0]['email'], $firstRow, 'Add user check - email');
+        $this->assertStringContainsString('(Re)send activation email', $firstRow, 'Add user check - activation text');
+    }
+
+    private function createArrayOfAddUsersFromArray($rolesArray)
+    {
+        foreach ($rolesArray as $key => $role) {
+            $email = 'add-user-'.$this->userRole.'-'.$role['roleName'].'-'.$this->testRunId.'@t.uk';
+            $this->expectedUsers[] = [
+                'email' => $email,
+                'firstName' => 'added',
+                'lastName' => 'user'.$role['roleName'].$this->userRole,
+                'postcode' => 'B'.strval($key).' 1ZZ',
+                'roleType' => $role['roleType'],
+                'role' => $role['role'],
+            ];
+            $this->userEmails[] = $email;
+        }
+    }
+
+    private function createArrayOfEditUsersFromArray($rolesArray)
+    {
+        foreach ($rolesArray as $key => $role) {
+            $email = 'edit-test-'.$role['roleName'].'-'.$this->testRunId.'@t.uk';
+            $this->expectedUsers[] = [
+                'email' => $email,
+                'firstName' => 'first'.str_replace('_', '', strtolower($role['role'])),
+                'lastName' => 'last'.str_replace('_', '', strtolower($role['role'])),
+                'postcode' => 'B'.strval($key).' 1YY',
+                'roleType' => $role['roleType'],
+                'role' => $role['role'],
+            ];
+            $this->userEmails[] = $email;
+        }
+    }
+
+    private function addUserInApplication()
+    {
+        $this->userCount = 0;
+        foreach ($this->expectedUsers as $addedUser) {
+            $this->iNavigateToAddNewUser();
+            $this->iAmOnAdminAddUserPage();
+            $this->checkRolesExistToAdd($this->userRoles);
+            $this->fillInAndSubmitUsers(
+                $addedUser['email'],
+                $addedUser['firstName'],
+                $addedUser['lastName'],
+                $addedUser['postcode'],
+                $addedUser['roleType'],
+                $addedUser['role']
+            );
+            ++$this->userCount;
+        }
+    }
+
+    /**
+     * @When I navigate to the add a new user page
+     */
+    public function iNavigateToAddNewUser()
+    {
+        $this->iAmOnAdminUsersSearchPage();
+        $this->clickBasedOnText('Add new user');
+    }
+
```

Comment:
This should probably live in `INavigateToAdminTrait` with its other navigations sibling functions.

Reply:
sneaky comment hiding away :)